package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -362,7 +363,20 @@ dhcp_add() {
 		DNS_SERVERS="$DNS_SERVERS $dnsserver"
 	}
 
-	append_bool "$cfg" ignore "--no-dhcp-interface=$ifname" && return 0
+	append_bool "$cfg" ignore "--no-dhcp-interface=$ifname" && {
+		if [ $ADD_LOCAL_GATENAME -gt 0 ] ; then
+			local dashedname="$( echo $ifname | sed 's/\./-/g' )"
+			local routername="$( uci_get system @system[0] hostname OpenWrt )"
```

Comment:
Hardcoding `OpenWrt` (or `LEDE`) here is not optimal. I suggest to source `/etc/os-release` and to use the `$ID` variable with a fall back to `OpenWrt`.

Example:

    local distribution="$( [ -f /etc/os-release ] && (. /etc/os-release; echo "$ID") || echo openwrt )"
    local routername="$( uci_get system @system[0] hostname "$distribution" )"